src/games/strategy/triplea/ai/fastAI/FastAI.java

```diff
@@ -9,14 +9,16 @@
  */
 public class FastAI extends ProAI {
 
+  // Odds estimator
+  private final static IOddsCalculator estimator = new FastOddsEstimator();
+
   public FastAI(final String name, final String type) {
     super(name, type);
   }
 
   @Override
-  public void initializeBattleCalculator() {
-    final IOddsCalculator calc = new FastOddsEstimator();
-    ProBattleUtils.setOddsCalculator(calc);
+  protected void initializeCalc() {
+    calc = new ProOddsCalculator(estimator);
```

Comment:
The inheritance structure I believe would be brittle. The 'is-a' relationship fails severaly. A 'FastAI' is not a ProAI, they make very different decisions, and similar to the other AIs. 

The inheritance hierarchy is a monolith. In the very rough diagram below, the small boxes are modules and the nested bigger boxes are classes that inherit from one another. Notice in this picture, if you change one of the submodules, the area of impact that is affected is very non-deterministic. You may or may not be altering the child classes, they may or may override stuff, and who knows if the child classes will work if you modify stuff in the parent classes.

![monolith](https://cloud.githubusercontent.com/assets/12397753/12691771/26749de8-c6a4-11e5-99e3-dae21890647e.png)

If you use composition on the other hand, generally achieved through IOC, then the submodules get to live on their own. They are then more re-usable, testable, and where they are used and what code they impact then becomes much more direct, and interactions become much more clear.

![ioc](https://cloud.githubusercontent.com/assets/12397753/12691804/6e662338-c6a4-11e5-9bfa-59041f689fb4.png)

So note, the relationship between any two entities is direct, one to one. In the nested inheritance class scenario, parent classes are related to child classes since children call the parents. The children are similarly related to the parents and the implementation modules of those parents which are once or twice removed from the children. IN effect, in the inheritance scenario, everything is related to everything else. IOC is nice because it helps decouple all of these entities, makes the interactions more straight forward and reduces them.